packages/daf-react-native-libsodium/src/key-management-system.ts

```diff
@@ -98,4 +98,9 @@ export class KeyManagementSystem extends AbstractKeyManagementSystem {
     debug('Deleting', kid)
     return await this.keyStore.delete(kid)
   }
+
+  async importKey(serializedKey: SerializedKey) {
+    await this.keyStore.set(serializedKey.kid, serializedKey)
```

Comment:
There should be some checks regarding the key capabilities.
Either here, or maybe at the `keyStore` level.
For example, if a `SerializedKey` is provided without a `privateKeyHex` component then it will be unusable after import.
Depending on the `keyStore` or the type of key other checks may be needed.

Reply:
Hi @mirceanis ,

This fix is for **daf-react-native-libsodium version 6** where the corresponding **daf-libsodium version 6** has the import function implemented in the exact fashion without any checks.

A code changes corresponds to version 7 can have a different PR.

Thanks

Reply:
ok, so this is not work in progress actually. It makes sense as a fix to daf v6.

I will be merging a bunch of PRs and it will be included in a next hotfix release for daf